hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -377,7 +378,25 @@ public void initTableMetadata() {
       if (initializeFromFilesystem(dataMetaClient, inflightInstantTimestamp)) {
         metrics.ifPresent(m -> m.updateMetrics(HoodieMetadataMetrics.INITIALIZE_STR, timer.endTimer()));
       }
+      return;
     }
+
+    // if metadata table exists, then check if any of the enabled partition types needs to be initialized
+    Set<String> inflightAndCompletedPartitions = getInflightAndCompletedMetadataPartitions(dataMetaClient.getTableConfig());
+    List<MetadataPartitionType> partitionsToInit = this.enabledPartitionTypes.stream()
+        .filter(p -> !inflightAndCompletedPartitions.contains(p.getPartitionPath()) && !MetadataPartitionType.FILES.equals(p))
+        .collect(Collectors.toList());
+
+    // if there are no partitions to initialize or there is a pending operation, then don't initialize in this round
+    if (partitionsToInit.isEmpty() || anyPendingDataInstant(dataMetaClient, inflightInstantTimestamp)) {
+      return;
+    }
+
+    String createInstantTime = getInitialCommitInstantTime(dataMetaClient);
```

Comment:
do we have a test for this. i.e. enable MDT files at commit 0 or commit5.. after few commits, enable col stats and after few commits enable bloom filter. query all 3 partitions, all of them should be intact.

Reply:
Added this test, but hitting a weird issue when bloom filter partition is being read:
```
java.lang.RuntimeException: file:/var/folders/s5/pqxf5ndx12qg6h0zgl2d9zxh0000gn/T/junit3683416470709715553/dataset/p1/4392e180-e6b1-4650-8ad0-cd8cf9f8cdb8_1-0-1_0000003.parquet is not a Parquet file. expected magic number at tail [80, 65, 82, 49] but found [0, 0, 0, 0]
```
Still looking into this issue.

Reply:
syned up f2f.. this is resolved. it was because HoodieTestTable does not really write records. Used write client for tests.